android/src/main/java/com/mybigday/rns3/RNS3TransferUtility.java

```diff
@@ -234,9 +235,42 @@ public void enableProgressSent(boolean enabled, Promise promise) {
   public void upload(ReadableMap options, Promise promise) {
     String bucket = options.getString("bucket");
     String key = options.getString("key");
+    String acl = options.getString("acl");
     File file = new File(options.getString("file"));
     ReadableMap meta = options.getMap("meta");
     ObjectMetadata metaData = new ObjectMetadata();
+    CannedAccessControlList _acl;
+
+    if (acl != null) {
+      switch (acl) {
+        case "private":
+          _acl = CannedAccessControlList.Private;
```

Comment:
Could we use [`CannedAccessControlList.valueOf(acl)`](http://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/services/s3/model/CannedAccessControlList.html#valueOf-java.lang.String-) instead of the `switch` statement? For upper camel case we can use [`uppercamelcase`](https://github.com/SamVerschueren/uppercamelcase) for `acl` in the JS function first.